src/ansys/dpf/core/helpers/streamlines.py

```diff
@@ -28,44 +29,173 @@ def __init__(self, data):
             )
         self._pv_data_set = None
-        self._streamlines_fc = None
+        self._streamlines_field = None
         if isinstance(data, pv.PolyData):
             self._pv_data_set = data
-        elif isinstance(FieldsContainer):
-            self._streamlines_fc = data
+        elif isinstance(data, dpf.Field):
+            self._streamlines_field = data
         else:
             raise AttributeError(
                 "streamlines must be a pyvista.PolyData or a dpf.FieldsContainer instance."
             )
 
-    def _pv_data_set_to_fc(self):
-        """Convert pyvista.PolyData into FieldsContainer."""
-        raise Exception("Not implemented yet")
+    def _set_vtk_cell_array(self, vtk_cell_array, vtk_poly_data):
+        raise Exception("_set_vtk_cell_array not implemented in child class")
+
+    def _cell_from_type(self, cell_type_id, vtk):
+        if cell_type_id == element_types.Line2.value:
+            vtk_cell = vtk.vtkLine()
+        elif cell_type_id == -4:
+            vtk_cell = vtk.vtkPolyLine()
+        elif element_types.Unknown.value:
+            vtk_cell = vtk.vtkGenericCell()
+        return vtk_cell
+
+    def _vtk_type_from_dpf_type(self, c, cell_types_converted, vtk):
+        if c == vtk.VTK_LINE:
+            cell_types_converted.append(element_types.Line2.value)
+        elif c == vtk.VTK_POLY_LINE:
+            cell_types_converted.append(int(-4))
+        elif c == vtk.VTK_POLY_VERTEX:
+            cell_types_converted.append(int(-3))
+        else:
+            cell_types_converted.append(element_types.Unknown.value)
+
+    def _pv_data_set_to_fc(self, server=None):
+        """Convert pyvista.PolyData into a Field."""
+        import vtk
+
+        data_set = self._pv_data_set
+        cell_points = []
+        cell_types = []
+        data_arrays = []
+        array_names_base = data_set.array_names
+        array_names = []
+        for n in array_names_base:
+            if not "streamlines" in n:
+                continue
+            array_names.append(n)
+            data_arrays.append(data_set[n])
+        for i in range(0, data_set.n_cells):
+            cell_points.append(data_set.cell_point_ids(i))
+            cell_types.append(data_set.cell_type(i))
+        points_array = data_set.points
+
+        # compute DPF objects
+        cell_types_converted = []
+        for c in cell_types:
+            self._vtk_type_from_dpf_type(c, cell_types_converted, vtk)
+        nodes_scoping = dpf.Scoping(location=locations.nodal, server=server)
+        nodes_scoping.ids = np.arange(1, data_set.n_points + 1)
+        streamlines_field = dpf.Field(location=locations.nodal, server=server)
+        fdef = streamlines_field.field_definition
+        if len(array_names) > 0:
+            fdef.name = array_names[0]
+        streamlines_field.scoping = nodes_scoping
+        if len(data_arrays) > 0:
+            streamlines_field.data = data_arrays[0]
```

Comment:
@anslpa If I understand correctly, what you are doing here is first doing a loop in all points in `self._pv_data_set` regardless of which streamline in particular they belong to, and then putting all of them in a very big field. What I would suggest is creating a `FieldsContainer`with one Field per streamline with label "streamline" (maybe?), and each one having a mesh support as the line mesh of the streamline. I expect that when we ship this as an operator it will we done in this way, as computing each streamline is independent from the other and all of them cuold be computed in parallel in a pragma omp for.

Reply:
After discussion with Rafael, we decided to return a FieldContainer with the following label : "line". Each line of the streamlines would be defined by a field. Each field would have a MeshedRegion associated with one cell element (multi lines cell) and all the connected points. 

Later, the "multi lines cell" needs to be handled by the PyVista plotter when running mesh_single_cell_multi_lines.plot()" (this might imply changes in the way we give the cells types to pyvista, in _vtk_helpers).